sdk/typescript/entrypoint/register.ts

```diff
@@ -114,15 +114,24 @@ function addArg(args: Arguments): (fct: Function_) => Function_ {
         description: arg.description,
       }
 
-      if (arg.defaultValue) {
-        opts.defaultValue = arg.defaultValue as string & { __JSON: never }
-      }
-
       let typeDef = addTypeDef(arg.type)
       if (arg.isOptional) {
         typeDef = typeDef.withOptional(true)
       }
 
+      // We do not set the default value if it's not a primitive type, we let typescript
+      // resolves the default value during the runtime instead.
+      // If it has a default value but is not primitive, we set the value as optional
+      // to workaround the fact that the API isn't aware of the default value and will
+      // expect it to be set as required input.
+      if (arg.defaultValue) {
+        if (isPrimitiveType(arg.type)) {
```

Comment:
Another way to do this is to try to serialize to JSON, and run the `else` block on the exception handling.

Reply:
I think it's less clear than checking the type

Reply:
Yes, but the problem is this is more limiting.